### PR TITLE
Create categories and datasets from json files on startup

### PR DESCRIPTION
**Pull request by matt-deboer**
This is an implementation of #50, allowing to automatically create categories and datasets on startup.

To create categories and datasets, set the environment variable `INITIALIZE_FROM_FILE` to the path of a json file which defines a set of categories and/or datasets to be created/updated on startup. 

For example:
```
{ 
  "categories": [
    { "name": "cat", "supercategory": "animal", "color": "#990000" },
    { "name":"dog", "supercategory": "animal", "color": "#5566AA"}
  ],
  "datasets": [
    {"name": "dataset1", "categories": ["dog", "cat"]}
  ]
}
```
Note: only the `name` attribute is required for a category or dataset.
Note: the net result is that the objects defined in this file are added and/or updated to the existing objects--but objects not listed in this file are ignored (i.e., not deleted).

### CODE REVIEW

File: app/models.py [hunk 1]
```diff
@@ -20,6 +21,23 @@ class DatasetModel(db.DynamicDocument):
...
+    def create_from_json(cls, json_file):
+        with open(json_file) as file:
+            datasets = json.load(file)
+            for dataset_json in datasets:
+                name = dataset_json.get('name')
+                if name:
+                    try:
+                        dataset = DatasetModel(
+                            name=name, 
+                            categories=dataset_json.get('categories', []))
```
Comment (jsbroks) [inline app/models.py]: Categories are stored as a list of ids (not names). You'll have to do an instance check, get the ids of named categories, and create any missing categories like [here](https://github.com/jsbroks/coco-annotator/blob/master/app/api/datasets.py#L66).

File: app/config.py [hunk 2]
```diff
@@ -12,5 +12,7 @@ class Config:
 
     TESTING = os.getenv("TESTING", False)
     DATASET_DIRECTORY = os.getenv("DATASET_DIRECTORY", "/datasets/")
-    LOAD_IMAGES_ON_START = False
+    CREATE_CATEGORIES_FROM_FILE = os.getenv("CREATE_CATEGORIES_FROM_FILE")
```
Comment (jsbroks) [inline app/config.py]: Is there a reason for two seperate files? I would perfer to have one with with a config property `Config.CREATE_FROM_FILE` (makes it simpler).

```javascript
{
  "categories": [
    { name: "cat", ...}
  ],
  "datasets": [
    { "name": "dataset", categories: ["a", "b", "c"] }
  ]
}
```

Comment (matt-deboer) [reply app/config.py]: no reason; one file sounds great

File: app/models.py [hunk 3]
```diff
@@ -185,6 +203,22 @@ def create_category(cls, name, color=None, metadata=None, supercategory=None):
         category.save()
         return category
 
+    @classmethod
+    def create_from_json(cls, json_file):
+        with open(json_file) as file:
+            categories = json.load(file)
+            for cat in categories:
```
[the remark this answers is not in the document]
Comment (matt-deboer) [reply app/models.py]: ok

**Comment by matt-deboer** [comment]
Ok @jsbroks, PTAL.
I've updated to a single file, and applied the other recommendations.

**Comment by SYLin117** [comment]
@matt-deboer Hi, thanks for adding this feature, but I have a problem with setup the INITIALIZE_FROM_FILE, I'm using this tool by docker-compose, after startup of the server, nothing happened. How exactly should I set up the path? any advice would be gratefull.

I place the file init.json at project root. and set  INITIALIZE_FROM_FILE = os.getenv('INITIALIZE_FROM_FILE', '/initializer.json') in config.py

**Pull request merged**